File: Programmers/크레인 인형뽑기 게임.py
def solution(board, moves):
    stack = []
    answer = 0
    for i in range(len(moves)):
        answer = pick_up(board, moves[i] - 1, stack, answer)

    return answer


def pick_up(board, pick, stack, answer):
    temp = 0
    for i in range(len(board)):
        if board[i][pick] != 0:
            temp = board[i][pick]
            board[i][pick] = 0
            break;

    if temp != 0:
        if len(stack) > 0 and stack[-1] == temp:
            stack.pop()
            answer += 2
        else:
            stack.append(temp)
    return answer

File: Programmers JS/크레인 인형뽑기 게임.js
function solution(board, moves) {
  console.log(board);
  let stack = [];
  let answer = 0;
  for (let i = 0; i < moves.length; i++) {
    let row = 0;
    let toy;
    while (row < board.length) {
      if (board[row][moves[i] - 1] === 0) {
        row++;
      } else {
        toy = board[row][moves[i] - 1];
        board[row][moves[i] - 1] = 0;
        break;
      }
    }
    if (stack.length > 0 && stack[stack.length - 1] === toy) {
      stack.pop();
      answer += 2;
    } else {
      if (toy !== undefined) {
        stack.push(toy);
      }
    }
  }

  return answer;
}
